File: javascript/verificar_numero.js
function esNumeroPositivo(valor){
    try{
        valor = parseInt(valor);
        if(valor > 0){
            return valor;            
        }else{
            alert("'" + String(valor) + "', no es positivo, intente nuevamente.");
            return false;
        }
            
        }catch{
            alert("'" + String(valor) + "', no es un número.")
            return false

        }
    }
    
    
   

   

File: problemas_algoritmos/python/verificar_numero.py
# funciones para verificar si una entrada es numero de lo contrario seguir solicitando la entrada
def esNumeroPositivo(valor):
    try:
        valor = int(valor)
        if valor > 0:
            return valor
        print("'" + str(valor) + "', no es positivo, intente nuevamente.")
        return False
    except:
        print("'" + str(valor) + "', no es un número.")
        return False

def solicitarNumeros(pregunta):
    respuesta = False
    while respuesta is False:
        respuesta = esNumeroPositivo(input(pregunta))
    return int(respuesta)